sentry_sdk/integrations/celery/__init__.py

```diff
@@ -433,3 +436,44 @@ def sentry_workloop(*args, **kwargs):
                     sentry_sdk.flush()
 
     Worker.workloop = sentry_workloop
+
+
+def _patch_producer_publish():
+    # type: () -> None
+    original_publish = Producer.publish
+
+    @ensure_integration_enabled(CeleryIntegration, original_publish)
+    def sentry_publish(self, *args, **kwargs):
+        # type: (Producer, *Any, **Any) -> Any
+        kwargs_headers = kwargs.get("headers", {})
+        if not isinstance(kwargs_headers, Mapping):
```

Comment:
If `kwargs_headers` is a list `[("key", "value"), ("k", "v")]` then it would be reset to an empty dict?

I am not sure if this list of tuples could happen, but I do not understand this if.

Reply:
Basically, what I am trying to do here is to make sure that the `get` operations (line 453 to 455) are successful. 

[`Mapping`](https://docs.python.org/3/library/collections.abc.html#collections.abc.Mapping) (full name `collections.abc.Mapping`) is the abstract base class that all [mapping types](https://docs.python.org/3/glossary.html#term-mapping) in Python inherit from. `dict` inherits from `Mapping`, so this `isinstance` is true for any dictionary, but it would also be possible to create a custom `Mapping` type, which provides similar functionality to a `dict`, but which does not inherit from `dict`. This `isinstance` also evaluates to `True` for any of these `dict`-like data structures.

Regarding your example, `isinstance([("key", "value"), ("k", "v")], Mapping)` would evaluate to `False` since `list` does not inherit from `Mapping` (it does not have a `get` method, which all `Mapping` types have). So, we would replace it with an empty dictionary to ensure that the `get` calls don't raise an exception.

Reply:
now I get it!